api/src/main/java/org/xnio/XnioWorker.java

```diff
@@ -874,6 +877,12 @@ protected final int getWorkerQueueSize() {
         return taskQueue.size();
     }
 
+    public abstract XnioWorkerMXBean getMetrics();
+
+    protected abstract Closeable registerServerMetrics(XnioServerMXBean metrics);
```

Comment:
This one is tricky, as yes it is closable as it closes mbean handles and everything.
but we still need to remove it from worker's server list, which you can only do by having reference to worker and remove it from the list.
If you have an idea how to change that, I dont mind changing that.